test/gtest/uct/ib/test_ib.cc

```diff
@@ -458,18 +493,23 @@ UCS_TEST_P(test_uct_ib_sl, check_ib_sl_config) {
     // go over all SLs, check UCTs could be initialized on a specific SL
     // and able to send/recv traffic
     for (sl = 0; sl < max_avail_sl; ++sl) {
+        bool sl_supports_ar = UCS_BIT_IS_SET(m_ooo_sl_mask, sl);
         if (!has_transport("rc_verbs") && !has_transport("ud_verbs")) {
             // if AR is configured on the given SL, set AR_ENABLE to "y",
             // otherwise - to "n" in order to test that AR_ENABLE parameter
             // works as expected w/o errors and warnings
-            modify_config("IB_AR_ENABLE",
-                          (m_ooo_sl_mask & UCS_BIT(sl)) ? "y" : "n");
+            modify_config("IB_AR_ENABLE", sl_supports_ar ? "y" : "n");
         }
+
         modify_config("IB_SL", ucs::to_string(static_cast<uint16_t>(sl)));
 
-        test_uct_ib::init();
-        send_recv_short();
-        test_uct_ib::cleanup();
+        for (int test_ddp = 0; test_ddp <= (has_ddp && sl_supports_ar);
+             ++test_ddp) {
+            set_ddp_enable(test_ddp);
+            test_uct_ib::init();
+            send_recv_short();
+            test_uct_ib::cleanup();
```

Comment:
this looks weird, need to create a new test class and instantiate it with several variants